pkg/common/vsphere/virtual_center_manager.go

```diff
@@ -0,0 +1,148 @@
+package vsphere
```

Comment:
Please take virtual_center_manager out, this is dangerous.

Each PE type is capable of taking different config parameters.  If we have IVD and VM it's both possible and desirable to have different logins/permissions for each type.  With this setup, the configs will silently fight with each other.

Singletons are a pain when you're trying to rework things/encapsulate.  There's no good need for a singleton here that I can see.

What is the value here?  If the idea is that changing one config will force the changes for all the other PE types, see my first comment.

Reply:
done, got rid of vcenter manager, thanks for pointing it out, it absolutely makes sense. Also got rid of other instances where singletons were being instantiated (vslm-manager and cns-manager)